In [1]:
import pandas as pd
import numpy as np
import json
import re

In [2]:
train = pd.read_csv("../data/raw/release_train_patients.csv")
train.head()

AGE                             DIFFERENTIAL_DIAGNOSIS SEX  \
0   18  [['Bronchitis', 0.19171203430383882], ['Pneumo...   M   
1   21  [['HIV (initial infection)', 0.518950056440760...   M   
2   19  [['Bronchitis', 0.11278064619119596], ['Pneumo...   F   
3   34  [['URTI', 0.23859396799565236], ['Cluster head...   F   
4   36  [['URTI', 0.23677812769175735], ['Influenza', ...   M   

                 PATHOLOGY                                          EVIDENCES  \
0                     URTI  ['E_48', 'E_50', 'E_53', 'E_54_@_V_161', 'E_54...   
1  HIV (initial infection)  ['E_9', 'E_27', 'E_50', 'E_51', 'E_53', 'E_54_...   
2                Pneumonia  ['E_53', 'E_54_@_V_179', 'E_54_@_V_192', 'E_55...   
3                     URTI  ['E_48', 'E_53', 'E_54_@_V_183', 'E_55_@_V_89'...   
4                     URTI  ['E_49', 'E_50', 'E_53', 'E_54_@_V_183', 'E_55...   

  INITIAL_EVIDENCE  
0             E_91  
1             E_50  
2             E_77  
3             E_53  
4            E_201

In [3]:
len(train['PATHOLOGY'].unique())

49

In [4]:
train['PATHOLOGY'].unique()

array(['URTI', 'HIV (initial infection)', 'Pneumonia',
       'Chronic rhinosinusitis', 'Viral pharyngitis', 'Anemia',
       'Atrial fibrillation', 'Allergic sinusitis', 'Larygospasm',
       'Cluster headache', 'Anaphylaxis', 'Spontaneous pneumothorax',
       'Acute pulmonary edema', 'Tuberculosis', 'Myasthenia gravis',
       'Panic attack', 'Scombroid food poisoning', 'Epiglottitis',
       'Inguinal hernia', 'Boerhaave', 'Pancreatic neoplasm',
       'Bronchitis', 'SLE', 'Acute laryngitis', 'Unstable angina',
       'Bronchiectasis', 'Possible NSTEMI / STEMI', 'Chagas',
       'Localized edema', 'Sarcoidosis', 'Spontaneous rib fracture',
       'GERD', 'Bronchospasm / acute asthma exacerbation',
       'Acute COPD exacerbation / infection', 'Guillain-Barré syndrome',
       'Influenza', 'Pulmonary embolism', 'Stable angina', 'Pericarditis',
       'Acute rhinosinusitis', 'Whooping cough', 'Myocarditis',
       'Acute dystonic reactions', 'Pulmonary neoplasm',
       'Acute otitis

In [5]:
train['PATHOLOGY'].value_counts()

URTI                                        64368
Viral pharyngitis                           61642
Anemia                                      50665
HIV (initial infection)                     29013
Localized edema                             27825
Anaphylaxis                                 27718
Pulmonary embolism                          27468
Influenza                                   26812
Bronchitis                                  26400
Allergic sinusitis                          26203
Acute dystonic reactions                    25982
GERD                                        25979
Acute otitis media                          25917
Pneumonia                                   25761
Panic attack                                25019
Acute laryngitis                            24129
Guillain-Barré syndrome                     22867
Pericarditis                                22785
Sarcoidosis                                 21285
Possible NSTEMI / STEMI                     21260


49 unique diseases - not too much

In [6]:
train.shape

(1025602, 6)

Million patients in train

In [7]:
with open("../data/raw/release_evidences.json") as file:
	data = json.load(file)

In [8]:
len(data)

223

In [9]:
B_type_data = {}
for i in data:
    if data[i]['data_type'] == "B":
        B_type_data[i] = data[i]

In [10]:
B_type_data

{'E_91': {'name': 'E_91',
  'code_question': 'E_91',
  'question_fr': 'Avez-vous objectivé ou ressenti de la fièvre?',
  'question_en': 'Do you have a fever (either felt or measured with a thermometer)?',
  'is_antecedent': False,
  'default_value': 0,
  'value_meaning': {},
  'possible-values': [],
  'data_type': 'B'},
 'E_53': {'name': 'E_53',
  'code_question': 'E_53',
  'question_fr': 'Avez-vous de la douleur à quelque part en lien avec votre raison de consultation?',
  'question_en': 'Do you have pain somewhere, related to your reason for consulting?',
  'is_antecedent': False,
  'default_value': 0,
  'value_meaning': {},
  'possible-values': [],
  'data_type': 'B'},
 'E_159': {'name': 'E_159',
  'code_question': 'E_159',
  'question_fr': 'Avez-vous perdu conscience?',
  'question_en': 'Did you lose consciousness?',
  'is_antecedent': False,
  'default_value': 0,
  'value_meaning': {},
  'possible-values': [],
  'data_type': 'B'},
 'E_129': {'name': 'E_129',
  'code_question': 'E_

223 evidences
208 with data type B (initial data for MVP)